File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int N=nums.length;
        int ans=Integer.MAX_VALUE, left=0, sum=0;
        
        for(int i=0; i<N; i++) {
            sum+=nums[i];
            
            while(sum>=target) {
                ans=Math.min(ans, i-left+1);
                sum-=nums[left];
                left++;
            }
        }
        
        return ans==Integer.MAX_VALUE ? 0 : ans;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
	ans, left, sum := math.MaxInt, 0, 0

	for i, num := range nums {
		// fmt.Printf("i= %d \t num=%d\n", i, num)
		sum += num
		for sum >= target {
			ans = min(ans, i-left+1)
			sum -= nums[left]
			left++
		}
	}

	if ans == math.MaxInt {
		return 0
	}
	return ans
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}
